src/views/listview/list-view.component.js

```diff
@@ -229,11 +232,42 @@ angular.module('patternfly.views').component('pfListView', {
       return ctrl.config.checkDisabled && ctrl.config.checkDisabled(item);
     };
 
+    function setPagination () {
+      if (angular.isUndefined(ctrl.pageConfig)) {
+        ctrl.pageConfig = {
+          pageNumber: 1,
+          pageSize: ctrl.items.length,
+          numTotalItems: ctrl.items.length,
+          showPaginationControls: false
+        };
+      } else {
+        if (angular.isUndefined(ctrl.pageConfig.showPaginationControls)) {
+          ctrl.pageConfig.showPaginationControls = true;
+        }
+        if (angular.isUndefined(ctrl.pageConfig.pageNumber)) {
+          ctrl.pageConfig.pageNumber = 1;
+        }
+        if (angular.isUndefined(ctrl.pageConfig.pageSize)) {
+          ctrl.pageConfig.pageSize = 10;
+        }
+        if (angular.isUndefined(ctrl.pageConfig.numTotalItems)) {
+          ctrl.pageConfig.numTotalItems = ctrl.items.length;
+        }
+        // if not showing pagination, keep pageSize equal to numTotalItems
+        if (!ctrl.pageConfig.showPaginationControls && ctrl.pageConfig.pageSize !== ctrl.pageConfig.numTotalItems) {
```

Comment:
Not following your comment?  This check is necessary to keep the pageSize = num TotalItems when not showing pagination controls (for example when filter is applied and removed).